src/dmd/root/outbuffer.d

```diff
@@ -386,6 +386,22 @@ struct OutBuffer
         writestring(unsignedToTempString(u, buf));
     }
 
+    /**************************************
+     * Convert `u` to a hex string and append it to the buffer.
+     * Params:
+     *  u = integral value to append
+     *  mw = min width with zero padding (e.g. u=255, mw=8 : '000000ff')
+     */
+    extern (C++) void printHex(ulong u, uint mw = 0) pure nothrow
```

Comment:
You can use a faster way to generate a hex String.
```
enum hexString = (ulong value)
{
    const wasZero = !value;
    static immutable NibbleRep = "0123456789abcdef";
    char[] resultBuffer;
    resultBuffer.length = 18; // ulong.sizeof * 2 + "0x".length
    resultBuffer[] = '0';
    int p;
    for(ubyte currentNibble = value & 0xF;
        value;
        currentNibble = ((value >>>= 4) & 0xF)
    )
    {
        resultBuffer[17 - p++] = NibbleRep[currentNibble];
    }
    resultBuffer[17 - wasZero - p++] = 'x';
    return cast(string) resultBuffer[17 - p - wasZero .. 18];
};
```

Reply:
@UplinkCoder, I saw your forum post about `unsignedToTempString` being slow, will you update it?

Reply:
at some point.